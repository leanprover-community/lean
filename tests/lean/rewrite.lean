class is_one_class (n : ℕ) :=
(one : n = 1)

lemma one_eq (n : ℕ) [is_one_class n] : 1 = n := is_one_class.one.symm

-- error message should say which typeclass is missing
example (n : ℕ) : n = 1 := by rw one_eq
