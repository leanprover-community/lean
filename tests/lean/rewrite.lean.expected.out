rewrite.lean:8:4: error: invalid rewrite tactic, failed to synthesize type class instance for is_one_class n
